Content.Shared/_DV/Projectiles/ProjectileForeignBodySystem.cs

```diff
@@ -0,0 +1,172 @@
+using Content.Shared._Shitmed.Targeting;
+using Content.Shared.Administration.Logs;
+using Content.Shared.Armor;
+using Content.Shared.Body.Part;
+using Content.Shared.Body.Systems;
+using Content.Shared.Database;
+using Content.Shared.EntityEffects;
+using Content.Shared.Inventory;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Projectiles;
+using Robust.Shared.Containers;
+using Robust.Shared.Network;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Shared._DV.Projectiles;
+
+public sealed class ProjectileForeignBodySystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+    [Dependency] private readonly SharedBodySystem _body = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly ISharedAdminLogManager _adminLog = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ProjectileForeignBodyComponent, ProjectileHitEvent>(OnEmbedProjectileHit);
+
+        SubscribeLocalEvent<SimpleForeignBodyContainerComponent, ComponentInit>(OnSimpleInit);
+        SubscribeLocalEvent<SimpleForeignBodyContainerComponent, ProjectileForeignBodyAttemptEvent>(OnSimpleEmbed);
+        SubscribeLocalEvent<SimpleForeignBodyContainerComponent, ForeignBodyEffectsEvent>(OnSimpleEffect);
+
+        SubscribeLocalEvent<BodyPartForeignBodyContainerComponent, ComponentInit>(OnBodyPartInit);
+        SubscribeLocalEvent<BodyForeignBodyContainerComponent, ProjectileForeignBodyAttemptEvent>(OnBodyEmbed);
+        SubscribeLocalEvent<BodyPartForeignBodyContainerComponent, ForeignBodyEffectsEvent>(OnBodyPartEffect);
+    }
+
+    private void OnEmbedProjectileHit(Entity<ProjectileForeignBodyComponent> ent, ref ProjectileHitEvent args)
+    {
+        if (!TryComp(ent, out ProjectileComponent? projectile))
+            return;
+
+        var evt = new CoefficientQueryEvent(SlotFlags.All);
+        RaiseLocalEvent(args.Target, evt);
+
+        var chance = ent.Comp.BaseChance * evt.DamageModifiers.Coefficients.GetValueOrDefault(ent.Comp.DamageType, 1);
+        if (_random.NextFloat() > chance)
+            return;
```

Comment:
it should do strong bleeding if it passes through you as exit wounds are really dangerous

Reply:
this would be akin to a flat buff on all bullets ever even in the non-lodged case which would be. something

Reply:
so then reduce pierce bleeding